diesel/src/pg/connection/stmt/mod.rs

```diff
@@ -41,16 +43,34 @@ impl Statement {
             .len()
             .try_into()
             .map_err(|e| crate::result::Error::SerializationError(Box::new(e)))?;
-        unsafe {
-            raw_connection.send_query_prepared(
-                self.name.as_ptr(),
-                param_count,
-                params_pointer.as_ptr(),
-                param_lengths.as_ptr(),
-                self.param_formats.as_ptr(),
-                1,
-            )
-        }?;
+
+        if let Some(ref sql) = self.sql {
+            // For regular unnamed statements, use send_query_params
+            unsafe {
+                raw_connection.send_query_params(
+                    sql.as_ptr(),
+                    param_count,
+                    param_types_to_ptr(self.param_types.as_ref()),
+                    params_pointer.as_ptr(),
+                    param_lengths.as_ptr(),
+                    self.param_formats.as_ptr(),
+                    1,
+                )
+            }?;
+        } else {
+            // For named statements, use the existing approach
```

Comment:
The expression "use the existing approach" looks reasonable now if we look at this change. We won't know what this means in a few month. So better put some longer comment above the if condition, explaining why this is necessary.